DashAI/back/core/schema_fields/component_field.py

```diff
@@ -0,0 +1,46 @@
+from pydantic import BaseModel, Field
+from typing_extensions import Annotated
+
+
+class ComponentType(BaseModel):
+    component: str
+    params: dict
+
+
+def component_field(
```

Comment:
Same here, you could use beartype to check if the parameter matches the specified types.

```suggestion
# from beartype import beartype
#....
@beartype
def component_field(
```